stdlib/public/Concurrency/Executor.swift

```diff
@@ -12,5 +12,59 @@
 
 import Swift
 
-// TODO: Custom Executors proposal will define these types
-//       https://forums.swift.org/t/support-custom-executors-in-swift-concurrency/44425/38
+/// A service which can execute jobs.
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public protocol Executor: AnyObject, Sendable {
+  func enqueue(_ job: UnownedJob)
+}
+
+/// A service which can execute jobs.
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public protocol SerialExecutor: Executor {
+  // This requirement is repeated here as a non-override so that we
+  // get a redundant witness-table entry for it.  This allows us to
+  // avoid drilling down to the base conformance just for the basic
+  // work-scheduling operation.
+  @_nonoverride
+  func enqueue(_ job: UnownedJob)
+
+  /// Convert this executor value to the optimized form of borrowed
+  /// executor references.
+  func asUnownedSerialExecutor() -> UnownedSerialExecutor
+}
+
+/// An unowned reference to a serial executor (a `SerialExecutor`
+/// value).
+///
+/// This is an optimized type used internally by the core scheduling
+/// operations.  It is an unowned reference to avoid unnecessary
+/// reference-counting work even when working with actors abstractly.
+/// Generally there are extra constraints imposed on core operations
+/// in order to allow this.  For example, keeping an actor alive must
+/// also keep the actor's associated executor alive; if they are
+/// different objects, the executor must be referenced strongly by the
+/// actor.
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+@frozen
+public struct UnownedSerialExecutor {
+  #if compiler(>=5.5) && $BuiltinExecutor
+  @usableFromInline
+  internal var executor: Builtin.Executor
+  #endif
+
+  @inlinable
+  internal init(_ executor: Builtin.Executor) {
+    #if compiler(>=5.5) && $BuiltinExecutor
+    self.executor = executor
+    #endif
+  }
+
+  @inlinable
+  public init<E: SerialExecutor>(ordinary executor: __shared E) {
```

Comment:
oh, this one I don't know, mind if I ask/learn about the implications of __shared? Is it a borrow?

Reply:
Yes, it's a borrow.  It's the default argument convention except on initializers.

Reply:
I see, thanks!